#! /bin/sh /usr/share/dpatch/dpatch-run
## 07_disable_netdict.dpatch by Andrew Lee <andrew@linux.org.tw>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch from RedHat's bugzilla provided by Tomas Hoger

@DPATCH@
diff -urNad stardict-3.0.1~/src/conf.cpp stardict-3.0.1/src/conf.cpp
--- stardict-3.0.1~/src/conf.cpp	2007-10-30 16:05:35.000000000 +0800
+++ stardict-3.0.1/src/conf.cpp	2009-07-12 19:59:22.336370606 +0800
@@ -52,7 +52,7 @@
 #else
 	add_entry("/apps/stardict/preferences/dictionary/use_custom_font", false);
 #endif
-	add_entry("/apps/stardict/preferences/network/enable_netdict", true);
+	add_entry("/apps/stardict/preferences/network/enable_netdict", false);
 	// Default stardictd server.
 	add_entry("/apps/stardict/preferences/network/server", std::string(_("dict.stardict.org")));
 	add_entry("/apps/stardict/preferences/network/port", 2628);
diff -urNad stardict-3.0.1~/src/prefsdlg.cpp stardict-3.0.1/src/prefsdlg.cpp
--- stardict-3.0.1~/src/prefsdlg.cpp	2007-10-30 16:14:07.000000000 +0800
+++ stardict-3.0.1/src/prefsdlg.cpp	2009-07-12 20:01:15.931967798 +0800
@@ -1037,6 +1037,10 @@
     GtkWidget *hbox1 = gtk_hbox_new(FALSE, 6);
     gtk_box_pack_start(GTK_BOX(hbox1),button,false,false,0);
     gtk_box_pack_start(GTK_BOX(vbox1),hbox1,false,false,0);
+
+    label = gtk_label_new(_("Warning: Requests to remote StarDict server are sent over the network in an unencrypted form. Do not enable this if you are translating sensitive documents."));
+    gtk_label_set_line_wrap(GTK_LABEL(label), TRUE);
+    gtk_box_pack_start(GTK_BOX(vbox1),label,false,false,0);
 }
 
 void PrefsDlg::on_setup_mainwin_searchWhileTyping_ckbutton_toggled(GtkToggleButton *button, PrefsDlg *oPrefsDlg)
